### PR TITLE
Fix: Implement get_boost_winners admin route

### LINKED ISSUE #324: Create New Admin Route `get_boost_winners`
## Description 📹

Implement a new admin route `get_boost_winners` that retrieves the list of winners for a specific boost. This route will allow admin users to fetch the winners based on the `boost_id` provided. It should be secured using the existing authentication middleware to ensure only authorized admin users can access this information.

## Proposed Actions 🛠️

Here’s a checklist of actions to follow for resolving this issue:

1. **Fork and Create Branch**:  
   Fork the repository and create a new branch using the issue number:

   ```bash
   git checkout -b fix-[issue-number]
   ```

2. **Implement Changes**:

   - **Create the Route**: add a new handler in `src/endpoints/admin/quest_boost`: `get_boost_winners.rs`

   - **Define Request Parameters**:

     ```rust
     pub_struct!(Deserialize; GetBoostWinnersParams {
         boost_id: i64,
     });
     ```

   - **Implement the Handler Logic, which should look something like that**:

     ```rust
     #[route(get, "/admin/boosts/get_boost_winners", auth_middleware)]
     pub async fn get_boost_winners_handler(
         State(state): State<Arc<AppState>>,
         Extension(_sub): Extension<String>, // Assuming admin authentication is handled
         Query(params): Query<GetBoostWinnersParams>,
     ) -> impl IntoResponse {
         let collection = state.db.collection::<BoostDocument>("boosts");

         let filter = doc! { "id": params.boost_id };

         match collection.find_one(filter, None).await {
             Ok(Some(boost_doc)) => (
                 StatusCode::OK,
                 Json(json!({ "winners": boost_doc.winner })).into_response(),
             ),
             Ok(None) => get_error(format!("Boost with id {} not found", params.boost_id)),
             Err(e) => get_error(format!("Error fetching boost winners: {}", e)),
         }
     }
     ```

3. **Run Tests and Commit Changes**:

   - **Testing**:
     - Verify the new route works as expected by testing with valid and invalid `boost_id` values.
     - Ensure that unauthorized access is correctly blocked by the `auth_middleware`.

   - **Commit Changes**:

     ```bash
     git commit -m "Fix: Implement get_boost_winners admin route"
     ```

## Required 📋

To keep our workflow smooth, please make sure you follow these guidelines:

- **Assignment**: Don't create a pull request if you weren’t assigned to this issue.
- **Timeframe**: Complete the task within **3 business days**.
- **Closing the Issue**: In your PR description, close the issue by writing `Close #[issue_id]`.
- **Review Process**:
  - Once you've submitted your PR, change the label to **"ready for review"**.
  - If changes are requested, address them and then update the label back to **"ready for review"** once done.
- **Testing**: Test your PR locally before pushing, and verify that tests and build are working after pushing.

Thank you for your contribution 🙏

⚠️ **WARNING**: Failure to follow the requirements above may result in being added to the OnlyDust blacklist, affecting your ability to receive future rewards.

### PR DESCRIPTION
**Pull request by omsant02**
Implements new GET endpoint `/admin/boosts/get_boost_winners` to retrieve winners for a specific boost ID. Protected by admin authentication middleware.

Close #324

### CODE REVIEW

File: src/endpoints/admin/quest_boost/mod.rs [hunk 1]
```diff
@@ -1,3 +1,6 @@
 pub mod create_boost;
+pub mod get_boost_winners;
 pub mod get_quest_users;
 pub mod update_boost;
+
+pub use get_boost_winners::get_boost_winners_handler;
```
Comment (Marchand-Nicolas) [inline src/endpoints/admin/quest_boost/mod.rs]: Why ?

File: src/endpoints/admin/quest_boost/get_boost_winners.rs [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+#[derive(Deserialize)]
+pub struct GetBoostWinnersParams {
+    boost_id: i64,
+}
+
+#[route(get, "/admin/boosts/get_boost_winners", auth_middleware)]
+pub async fn get_boost_winners_handler(
+    State(state): State<Arc<AppState>>,
+    Extension(_sub): Extension<String>, // Assuming admin authentication is handled
```
Comment (Marchand-Nicolas) [inline src/endpoints/admin/quest_boost/get_boost_winners.rs]: You can remove this comment

**Review by Marchand-Nicolas — CHANGES_REQUESTED** [review]

**Comment by omsant02** [comment]
i have pushed the changes @Marchand-Nicolas 
please review it

Thank you

**Review by Marchand-Nicolas — APPROVED** [review]
Perfect, lgtm!

**Pull request merged**